system/gatekeeper/templates/constrainttemplate-restrict-monsoon3-namespace.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: gkrestrictmonsoon3namespace
+spec:
+  crd:
+    spec:
+      names:
+        kind: GkRestrictMonsoon3Namespace
+      validation:
+        openAPIV3Schema:
+          type: object
+          properties:
+            helmManifestParserURL:
+              type: string
+            patterns:
+              type: object
+              properties:
+                monsoon3:
+                  type: string
+                monsoon3global:
+                  type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      libs:
+        - |
+          {{ .Files.Get "lib/helm-release.rego" | nindent 10 }}
+      rego: |
+        package restrictnewnamespace
+        import data.lib.helm_release
+        
+        iro := input.review.object
+        n := iro.metadata.namespace
+        release := helm_release.parse_k8s_object(iro, input.parameters.helmManifestParserURL)
```

Comment:
Parsing is very expensive and will not be needed here. We can match on the secret name only.